### PR TITLE
introducing react query to project

### PR DESCRIPTION
**Pull request by RRamirez1414**

### CODE REVIEW

File: src/components/SearchPage.tsx [hunk 1]
```diff
@@ -3,36 +3,26 @@ import Card from './Card'
...
   })
-  const [responseData, setResponseData] = useState<ListCardsResponse>({
-    data: [],
-    page: 1,
-    pageSize: 1,
-    count: 1,
-    totalCount: 0,
-  })
+  
+  const { isLoading, isError, data } = useQuery<ListCardsResponse, Error>(
```
Comment (m1yon) [inline src/components/SearchPage.tsx]: it's best practice to not destructure these variables actually, if there's multiple queries in 1 component it becomes a nightmare quickly if you're destructuring all these different variables.

File: src/components/SearchPage.tsx [hunk 2]
```diff
@@ -3,36 +3,26 @@ import Card from './Card'
 import Pagination from './Pagination'
 import { fetchCards, ListCardsResponse } from 'utils'
 import { useFormInputDebounce } from 'hooks'
+import { useQuery } from 'react-query'
+
+const getCards = (name: string, currentPage: number) => {
```
Comment (m1yon) [inline src/components/SearchPage.tsx]: I would just define this inline, no need to make this a seperate function

File: src/components/SearchPage.tsx [hunk 3]
```diff
@@ -82,39 +69,39 @@ const SearchPage = () => {
             value={formData.pokemonName}
             ref={inputRef}
           />
-          {!isLoaded ? <div className="loader"></div> : null}
+          {isLoading && <div className="loader"></div>}
```
Comment (m1yon) [inline src/components/SearchPage.tsx]: I would always do it the other way:

```suggestion
          {!isLoaded ? <div className="loader"></div> : null}
```

Check out [this article](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx).

**Review by m1yon — CHANGES_REQUESTED** [review]

File: src/components/SearchPage.tsx [hunk 4]
```diff
@@ -17,13 +13,14 @@ const SearchPage = () => {
   const [formData, setFormData] = useState({
     pokemonName: '',
   })
-  
-  const { isLoading, isError, data } = useQuery<ListCardsResponse, Error>(
+  // { isLoading, isError, data }
```
Comment (RRamirez1414) [inline src/components/SearchPage.tsx:16]: comment was removed after push

**Review by m1yon — APPROVED** [review]

**Pull request merged**